File: code/21.go
/*
 * @Author: lmio
 * @Date: 2023-02-10 16:45:35
 * @LastEditTime: 2023-02-19 23:26:17
 * @FilePath: /leetcode/code/21.go
 * @Description:21.合并两个有序链表
 */
package code


func MergeTowList(list1 *ListNode, list2 *ListNode) *ListNode {
	head := &ListNode{}
	node := head
	for list1 != nil && list2 != nil {
		if list1.Val < list2.Val {
			node.Next = list1
			list1 = list1.Next
		}else {
			node.Next = list2
			list2 = list2.Next
		}
		node = node.Next
	}
	if list1 == nil {
		node.Next = list2
	}else {
		node.Next = list1
	}
	return head.Next
}

File: code/21.py
'''
Author: lmio 2091319361@qq.com
Date: 2023-08-05 12:25:44
LastEditors: lmio 2091319361@qq.com
Description:21. 合并两个有序链表 
'''

from typing import Optional

from utils.node import ListNode


class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode()
        cur = dummy
        while list1 and list2:
            if list1.val < list2.val:
                cur.next = list1
                list1 = list1.next
            else:
                cur.next = list2
                list2 = list2.next
            cur = cur.next
        if list1:
            cur.next = list1
        else:
            cur.next = list2
        return dummy.next